third_party/move/move-compiler-v2/src/env_pipeline/inliner.rs

```diff
@@ -1070,9 +1197,19 @@ impl<'env, 'rewriter> ExpRewriterFunctions for InlinedRewriter<'env, 'rewriter>
             _ => false,
         };
 
+        if let ExpData::SpecBlock(_, _) = exp.as_ref() {
+            self.in_spec += 1;
+        } else if self.in_spec > 0 {
+            self.in_spec += 1;
```

Comment:
Not obvious why we increment `in_spec` in this `else if`. Can you explain?

Reply:
It is because rewrite_exp will be called multiple times during visit of spec block, if we don't increase it each time, in_spec will be reset to 0 incorrectly.